src/quoting/spline_pool.rs

```diff
@@ -0,0 +1,487 @@
+            ]
+        } else {
+            vec![] // Empty ticks for zero liquidity
+        };
+
+        SplinePool {
+            base_pool: BasePool::new(
+                NodeKey {
+                    token0,
+                    token1,
+                    fee: 0,
+                    tick_spacing: MAX_TICK_SPACING,
+                    extension,
+                },
+                BasePoolState {
+                    sqrt_ratio,
+                    liquidity: liquidity_factor.into(),
+                    active_tick_index: if liquidity_factor > 0 {
+                        Some(0)
+                    } else {
+                        None
+                    },
+                },
+                ticks,
+            ),
+            liquidity_factor,
+            total_shares: liquidity_factor.into(),
+            last_compound_time,
+        }
+    }
+
+    fn calculate_shares(&self, total_shares: u256, factor: u128, total_factor: u128) -> u256 {
+        if total_factor == 0 {
+            return factor.into(); // First liquidity addition
+        }
+        let denom: u256 = total_factor.into();
+        let num: u256 = factor.into();
+        (total_shares * num) / denom
+    }
+    
+    fn calculate_factor(&self, total_factor: u128, shares: u256, total_shares: u256) -> u128 {
+        if total_shares == 0.into() {
+            return 0; // No shares exist
+        }
+        let total_factor_u256: u256 = total_factor.into();
+        let factor_u256 = (total_factor_u256 * shares) / total_shares;
+        factor_u256.try_into().unwrap()
+    }
+
+    pub fn compound_fees(&mut self, block_time: u64) -> u128 {
+        if block_time <= self.last_compound_time {
+            return 0; // No time has passed, no fees to compound
+        }
+        
+        // In the real contract, this would calculate and add fees
+        // For simulation, we just track that compounding has occurred
+        // by updating the timestamp
+        let liquidity_fees = 0; // No actual fee calculation in the simplified model
+        self.last_compound_time = block_time;
+        
+        liquidity_fees
+    }
+}
+
+impl Pool for SplinePool {
+    type Resources = SplinePoolResources;
+    type State = SplinePoolState;
+    type QuoteError = BasePoolQuoteError;
+    type Meta = BlockTimestamp;
+
+    fn get_key(&self) -> &NodeKey {
+        self.base_pool.get_key()
+    }
+
+    fn get_state(&self) -> Self::State {
+        SplinePoolState {
+            base_pool_state: self.base_pool.get_state(),
+            liquidity_factor: self.liquidity_factor,
+        }
+    }
+
+    fn quote(
+        &self,
+        params: QuoteParams<Self::State, Self::Meta>,
+    ) -> Result<Quote<Self::Resources, Self::State>, Self::QuoteError> {
+        let block_time = params.meta;
+        
+        // Check if fee compounding would happen in this quote
+        let should_compound = params.override_state.is_none() && block_time > self.last_compound_time;
```

Comment:
Spline pools only compound on `spline_v0::lp::add_liquidity`, `spline_v0::lp::remove_liquidity`, so won't need to calculate this on swap I don't think -- can ignore the possibility of compound when quoting swap.